spark/src/main/kotlin/com/adevinta/spark/SparkTheme.kt

```diff
@@ -0,0 +1,244 @@
+    ) {
+        MaterialTheme(
+            colorScheme = rememberedColors.asMaterial3Colors(),
+            typography = typo.asMaterial3Typography(),
+            shapes = internalShapes.asMaterial3Shapes(),
+        ) {
+            CompositionLocalProvider(
+                LocalContentColor provides SparkTheme.colors.onSurface,
+            ) {
+                ProvideTextStyle(value = SparkTheme.typography.body) {
+                    content()
+                }
+            }
+        }
+    }
+}
+
+@Suppress("ModifierMissing") // It's okay since its a base theme
+@Composable
+fun PreviewTheme(
+    themeVariant: ThemeVariant = if (LocalInspectionMode.current && isSystemInDarkTheme()) {
+        ThemeVariant.Dark
+    } else {
+        ThemeVariant.Light
+    },
+    userType: UserType = UserType.Part,
+    padding: PaddingValues = PaddingValues(16.dp),
+    contentPadding: Dp = 16.dp,
+    content: @Composable ColumnScope.() -> Unit,
+) {
+    SparkTenantTheme(
+        // We don't want to automatically support dark theme in the app but still want it in the previews
+        useDarkColors = themeVariant == ThemeVariant.Dark,
+        isPro = userType == UserType.Pro,
+    ) {
+        val previewContent = movableContentOf {
+            Column(
+                modifier = Modifier.padding(padding),
+                verticalArrangement = Arrangement.spacedBy(contentPadding),
+            ) {
+                content()
+            }
+        }
+        Surface(
+            color = SparkTheme.colors.surface,
+            content = previewContent,
+        )
+    }
+}
+
+@Composable
+internal fun SparkTenantTheme(
+    // We don't want to automatically support dark theme in the app but still want it in the previews
+    useDarkColors: Boolean = isSystemInDarkTheme(),
+    useSparkTokensHighlighter: Boolean = false,
+    useSparkComponentsHighlighter: Boolean = false,
+    isPro: Boolean = false,
+    content: @Composable () -> Unit,
+) {
+    val colors = if (useDarkColors) {
+        darkSparkColors(isPro = isPro)
+    } else {
+        lightSparkColors(isPro = isPro)
+    }
+    SparkTheme(
+        colors = colors,
+        shapes = sparkShapes(),
+        typography = sparkTypography(),
+        useSparkTokensHighlighter = useSparkTokensHighlighter,
+        useSparkComponentsHighlighter = useSparkComponentsHighlighter,
+        isPro = isPro,
+        content = content,
+    )
+}
+
+/**
+ * Contains functions to access the current theme values provided at the call site's position in
+ * the hierarchy.
+ */
+object SparkTheme {
+    /**
+     * Retrieves the current [SparkColors] at the call site's position in the hierarchy.
+     */
+    val colors: SparkColors
+        @Composable
+        @ReadOnlyComposable
+        get() = LocalSparkColors.current
+
+    /**
+     * Retrieves the current [SparkTypography] at the call site's position in the hierarchy.
+     *
+     */
+    val typography: SparkTypography
```

Comment:
I'm wondering why this type is in the singular form but not the 2 others.

Reply:
We used the same names as Material to keep the same terminology 🥲

Reply:
ack